### PR TITLE
Update `pandas` to 2.0.0

### PR DESCRIPTION
**Pull request by jtpio**
### Description

Try updating to the latest pandas which was just released on PyPI: https://pypi.org/project/pandas/2.0.0

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

**Review by hoodmane — COMMENTED** [review]
Thanks @jtpio!

**Comment by ryanking13** [comment]
Cool! I guess we need to try building pyarrow (#2933), since many people will now ask "how can I use pyarrow in pandas 2.0" :)

File: packages/pandas/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 package:
   name: pandas
-  version: 1.5.3
+  version: 2.0.0
   tag:
     - min-scipy-stack
   top-level:
     - pandas
 source:
   url: https://files.pythonhosted.org/packages/74/ee/146cab1ff6d575b54ace8a6a5994048380dc94879b0125b25e62edcb9e52/pandas-1.5.3.tar.gz
```
[the remark this answers is not in the document]
Comment (jtpio) [reply packages/pandas/meta.yaml]: Right, thanks!

**Comment by hoodmane** [comment]
Looks like geopandas hasn't released a version supporting Pandas 2.0 yet.

**Comment by hoodmane** [comment]
https://github.com/geopandas/geopandas/issues/2692

**Comment by bollwyvl** [comment]
Looks like `geopandas` supports pandas 2.0 now, perhaps worth revisiting?

**Comment by hoodmane** [comment]
Okay I opened a PR attempting the update. @ryanking13 reverted Geopandas 0.13 in #4088 so it may be a nontrivial upgrade...

**Comment by rootsmusic** [comment]
@hoodmane Your PR didn't update pandas?

**Comment by luiscastro193** [comment]
Is something blocking this?

**Comment by ryanking13** [comment]
Nop, we already have pandas==2.2.0 in the main branch so this can be closed. Note that we don't have pyarrow in pandas though.

**Pull request closed without merging**